internal/controller/http/correlation/middleware.go

```diff
@@ -27,25 +30,25 @@ func ManageHeader(next http.Handler) http.Handler {
 	})
 }
 
-func OnResponseComplete(next http.Handler) http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		begin := time.Now()
-		next.ServeHTTP(w, r)
-		correlationId := IdFromContext(r.Context())
-		lc := LoggingClientFromContext(r.Context())
-		if lc != nil {
+func LoggingMiddleware(lc logger.LoggingClient) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			begin := time.Now()
```

Comment:
Should we check if log level is TRACE and return early if not to be more efficient?